pkg/streamingpromql/types/limiting_pool.go

```diff
@@ -200,7 +219,14 @@ func (p *LimitingBucketedPool[S, E]) Put(s S, tracker *limiting.MemoryConsumptio
 		}
 	}
 
-	tracker.DecreaseMemoryConsumption(uint64(cap(s)) * p.elementSize)
+	estimatedBytes := uint64(cap(s)) * p.elementSize
+	if sm, isSeriesMetadata := any(s).(SeriesMetadata); isSeriesMetadata {
+		for _, l := range sm.Labels {
+			estimatedBytes += uint64(len(l.Name)+len(l.Value)) * StringSize
+		}
+	}
```

Comment:
I think we should tackle incorporating the label memory consumption in a separate PR - this will require a number of changes in a number of places, not just in this pools like this. 

It may also be worthwhile raising a Prometheus PR to add something like a `ByteSize()` method to `labels.Labels`, so we can cheaply get the size of a `labels.Labels` without iterating over all the labels like this.

Reply:
@charleskorn I will add this in separate PR, but can you ellaborate more how adding such method in Prometheus can make it cheap? Won't be at the end still have to iterate the slice anyway?

Reply:
> Won't be at the end still have to iterate the slice anyway?

No: we can just return the length of the underlying byte slice when `stringlabels` is in use.